server/src/main/java/io/littlehorse/common/model/Getable.java

```diff
@@ -171,6 +175,90 @@ public abstract List<IndexedField> getIndexValues(
         String key,
         Optional<TagStorageTypePb> tagStorageTypePb
     );
+
+    public List<Tag> getIndexEntries() {
+        List<Tag> out = new ArrayList<>();
+        for (GetableIndex<? extends Getable<?>> indexConfiguration : this.getIndexConfigurations()) {
+            if (!indexConfiguration.isValid(this)) {
+                continue;
+            }
+            Optional<TagStorageTypePb> tagStorageTypePb = indexConfiguration.getTagStorageTypePb();
+            List<IndexedField> singleIndexedValues = indexConfiguration
```

Comment:
Additionally there are ways to split streams (or branching streams) https://www.baeldung.com/java-split-stream

Reply:
Good points, this will be simplified in the next PR. I moved it over because I'm going to delete `getableStorageManager` (actually, we already did in the current PR).